nuitka/build/static_src/CompiledCodeHelpers.c

```diff
@@ -541,6 +541,44 @@ PyObject *BUILTIN_LEN(PyObject *value) {
     return PyInt_FromSsize_t(res);
 }
 
+PyObject *BUILTIN_ANY(PyObject *value) {
+    CHECK_OBJECT(value);
+
+    PyObject *it = PyObject_GetIter(value);
+    PyObject *(*iternext)(PyObject *);
+    PyObject *item;
+
+    if (unlikely((it == NULL)))
+        return NULL;
+    iternext = *Py_TYPE(it)->tp_iternext;
+
+    int cmp;
+
+    for (;;) {
+        item = iternext(it);
+        if (unlikely((item == NULL)))
+            break;
+        cmp = PyObject_IsTrue(item);
+        Py_DECREF(item);
+        if (unlikely(cmp < 0)) {
+            Py_DECREF(it);
+            return NULL;
+        }
+        if (cmp > 0) {
+            Py_DECREF(it);
+            Py_INCREF(Py_True);
+            return Py_True;
+        }
+    }
+    Py_DECREF(it);
+    if (CHECK_AND_CLEAR_STOP_ITERATION_OCCURRED())
```

Comment:
Need to checkout what this function `CHECK_AND_CLEAR_STOP_ITERATION_OCCURRED` does.

 believe the handling of the return value is definitely incorrect. Ok, the documentation of it is really not telling a lot, so please extract and add a better one from this.

It returns true, if no error occured. In CPython, the `iternext` is allowed to return `NULL` with no error set, and that means end of iteration. Alternatively, the CPython exception can be set as `StopIteration`, which would be wasteful, but allowed. In these 2 cases it returns true, and the loop is properly terminated. I guess, in this case, we didn't find a true value from `PyObject_IsTrue`, so you ought to return `Py_False ` in that case.

When it returns false, the loop iteration aborted with another error, but `StopIteration`, e.g. `KeyError` of the `next` of an iteration, that must be returned. So `return FALSE` should be done.

And actually, removing the `PyErr_Clear()` that ought to be harmless (clearing no exception, because none will the set at that point).

BTW, loop exit without exception should be considered unlikely, therefore I would prefer a condition like this:

```
          if (unlikely(!CHECK_AND_CLEAR_STOP_ITERATION_OCCURRED())) {
              return NULL;
         }
```

Just to make it clear, that this is going to be rare.